BitwardenShared/Core/Autofill/Services/AutofillCredentialService.swift

```diff
@@ -225,6 +244,85 @@ extension DefaultAutofillCredentialService: AutofillCredentialService {
 
         return ASPasswordCredential(user: username, password: password)
     }
+
+    @available(iOS 17.0, *)
+    func provideFido2Credential( // swiftlint:disable:this function_body_length
+        for passkeyRequest: ASPasskeyCredentialRequest,
+        withUserInteraction: Bool,
+        fido2UserVerificationMediatorDelegate: Fido2UserVerificationMediatorDelegate
+    ) async throws -> ASPasskeyAssertionCredential {
+        guard let credentialIdentiy = passkeyRequest.credentialIdentity as? ASPasskeyCredentialIdentity else {
+            throw AppProcessorError.invalidOperation
+        }
+
+        let isLocked = try? await authRepository.isLocked()
+        let vaultTimeout = try? await vaultTimeoutService.sessionTimeoutValue(userId: nil)
+
+        switch (vaultTimeout, isLocked) {
+        case (.never, true):
+            // If the user has enabled Never Lock, but the vault is locked,
+            // unlock the vault before continuing.
+            try await authRepository.unlockVaultWithNeverlockKey()
+        case (_, false):
+            break
+        default:
+            if !withUserInteraction {
+                throw Fido2Error.userInteractionRequired
```

Comment:
❓ If you get to this point and the vault is locked, isn't this method ultimately going to fail whether or not user interaction is available? It seems like we could just throw an error here without needing the `withUserInteraction` flag?

Reply:
🤔 That's a very good point. If I'm not mistaken there is no way this method gets called if the vault is locked with user interaction. So I think I could remove the `withUserInteraction` parameter. Nice catch!!!! 🎉